source/Firmata/UwpFirmata.cpp

```diff
@@ -333,6 +348,35 @@ UwpFirmata::inputThread(
     }
 }
 
+void
+UwpFirmata::onConnectionEstablished(
+    void
+    )
+{
+    {   //critical section guarantees state is only changed when it is not being modified elsewhere
```

Comment:
that is the intention, as this class is not designed to be stateful & the 'readiness' of UwpFirmata only depends on having a valid transport layer. As the attached IStream object fires its OnConnectionEstablished event, this class can declare it is ready for action. 
_connection_ready is a std::atomic_bool, so doesn't need interlocked operations. The lock here is intended to verify that we are the sole owner of the object while we change the state